django/db/models/fields/related.py

```diff
@@ -2051,6 +2052,18 @@ def db_type(self, connection):
     def db_parameters(self, connection):
         return {"type": self.db_type(connection), "check": []}
 
+    def get_db_converters(self, connection):
+        backend_converters = connection.ops.get_db_converters(Col(self.attname, self.related_field))
+        backend_converters = [self._convert_backend_converter(func) for func in backend_converters]
+        field_converters = self.related_field.get_db_converters(connection)
+        self_converters = super(ForeignKey, self).get_db_converters(connection)
```

Comment:
An equivalent, somewhat less ugly option: Instead of the above, add a `from_db_value` method:
```python
    def from_db_value(self, value, connection, context):
        if ((not value) and isinstance(value, six.string_types) and 
            connection.features.interprets_empty_strings_as_nulls):
                return None 
        return value
```
`super(ForeignKey, self).get_db_converters(connection)`  then invokes it. But I'm really not sure that's what `from_db_value` was intended for.